transformers/storage/median/repository.go

```diff
@@ -0,0 +1,286 @@
+		}
+		return insertErr
+	}
+	return tx.Commit()
+}
+
+func getOrclAddress(keys map[types.Key]string) (string, error) {
+	user, ok := keys[constants.Address]
+	if !ok {
+		return "", types.ErrMetadataMalformed{MissingData: constants.A}
+	}
+	return user, nil
+}
+
+func (repository *MedianStorageRepository) insertBud(diffID, headerID int64, metadata types.ValueMetadata, contractAddress, bud string) error {
+	budAddress, budErr := getBudAddress(metadata.Keys)
+	if budErr != nil {
+		return budErr
+	}
+
+	tx, txErr := repository.db.Beginx()
+	if txErr != nil {
+		return txErr
+	}
+
+	addressID, addressErr := shared.GetOrCreateAddress(contractAddress, repository.db)
+	if addressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("contract address", addressErr)
+		}
+		return addressErr
+	}
+
+	budAddressID, budAddressErr := shared.GetOrCreateAddress(budAddress, repository.db)
+	if budAddressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("median bud address", budAddressErr)
+		}
+		return addressErr
+	}
+
+	_, insertErr := tx.Exec(insertMedianBudQuery, diffID, headerID, addressID, budAddressID, bud)
+	if insertErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("bud record with address", insertErr)
+		}
+		return insertErr
+	}
+	return tx.Commit()
+}
+
+func getBudAddress(keys map[types.Key]string) (string, error) {
+	user, ok := keys[constants.A]
+	if !ok {
+		return "", types.ErrMetadataMalformed{MissingData: constants.A}
+	}
+	return user, nil
+}
+
+func (repository *MedianStorageRepository) insertSlot(diffID, headerID int64, metadata types.ValueMetadata, contractAddress, slot string) error {
+	slotID, slotErr := getSlotID(metadata.Keys)
+	if slotErr != nil {
+		return slotErr
+	}
+
+	tx, txErr := repository.db.Beginx()
+	if txErr != nil {
+		return txErr
+	}
+
+	addressID, addressErr := shared.GetOrCreateAddress(contractAddress, repository.db)
+	if addressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("contract address", addressErr)
+		}
+		return addressErr
+	}
+
+	slotAddressID, slotAddressErr := shared.GetOrCreateAddress(slot, repository.db)
+	if slotAddressErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("median slot address", slotAddressErr)
+		}
+		return slotAddressErr
+	}
+
+	_, insertErr := tx.Exec(insertMedianSlotQuery, diffID, headerID, addressID, slotID, slotAddressID)
+	if insertErr != nil {
+		rollbackErr := tx.Rollback()
+		if rollbackErr != nil {
+			return shared.FormatRollbackError("slot record with address", insertErr)
+		}
+		return insertErr
+	}
+	return tx.Commit()
+}
+
+func getSlotID(keys map[types.Key]string) (int, error) {
+	slotID, err := strconv.Atoi(keys[constants.SlotId])
```

Comment:
might be good to derive the string representation of the slot id with
```
slotIDstr, ok := keys[constants.SlotId]
```
so that we can also return an error if the key is missing